modern/src/reports/RouteReportPage.js

```diff
@@ -26,35 +27,62 @@ const Filter = ({ setItems }) => {
 };
 
 const RouteReportPage = () => {
+  const distanceUnit = useAttributePreference('distanceUnit');
+
+  const columns = [{
+    headerName: t('positionFixTime'),
+    field: 'fixTime',
+    type: 'dateTime',
+    flex: 1,
+    valueFormatter: ({ value }) => formatPosition(value, 'fixTime'),
+  }, {
+    headerName: t('positionLatitude'),
+    field: 'latitude',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatPosition(value, 'latitude'),
+  }, {
+    headerName: t('positionLongitude'),
+    field: 'longitude',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatPosition(value, 'longitude'),
+  }, {
+    headerName: t('positionSpeed'),
+    field: 'speed',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatPosition(value, 'speed'),
+  }, {
+    headerName: t('positionAddress'),
+    field: 'address',
+    type: 'string',
+    flex: 1,
+    valueFormatter: ({ value }) => formatPosition(value, 'address'),
+  }, {
+    headerName: t('positionIgnition'),
+    field: 'ignition',
+    type: 'boolean',
+    flex: 1,
+    valueGetter: ({ getValue }) => getValue('attributes').ignition ? 'Yes' : 'No',
```

Comment:
I think you misunderstood how it supposed to work. You need both `valueGetter` and `valueFormatter`. One gives you raw data and then another one formats it for display. Also, using `getValue` in `valueGetter` kind of defeats the purpose, I think.

Reply:
According to Material UI docs-

field is the only required property since it's the column identifier. It's also used to match with GridRowData values.

About valueGetter - Sometimes a column might not have a corresponding value and you just want to render a combination of different fields. To do that, you can set the valueGetter attribute of GridColDef as in the example below:

Note: You need to set a sortComparator for the column sorting to work when setting the valueGetter attribute.

https://material-ui.com/components/data-grid/rendering/#header-cell

Reply:
In our case, we don't have totalDistance and ignition as direct fields but these are the properties of attributes. I think I have to implement it in this way only.

Reply:
In our case we don't have direct fields, but we do have values. We just need proper getters for those values.